### PR TITLE
Nessie: Strip trailing slash for warehouse location

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Other catalogs does the same and for some file system double slash in a path creates the problem.

### CODE REVIEW

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 1]
```diff
@@ -183,7 +184,8 @@ private String validateWarehouseLocation(String name, Map<String, String> catalo
           catalogOptions);
       throw new IllegalStateException("Parameter 'warehouse' not set, Nessie can't store data.");
     }
-    return warehouseLocation;
+
+    return LocationUtil.stripTrailingSlash(warehouseLocation);
```
Comment (ajantha-bhat) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:188]: Just making this inline with other catalogs. 
We can lookup the callers of `LocationUtil.stripTrailingSlash`

**Comment by ajantha-bhat** [comment]
cc: @snazy, @dimas-b , @adutra

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java [hunk 2]
```diff
@@ -160,4 +162,15 @@ protected boolean supportsServerSideRetry() {
   public void testConcurrentCreateTransaction() {
     super.testConcurrentCreateTransaction();
   }
+
+  @Test
+  public void testWarehouseLocationWithTrailingSlash() {
+    Assertions.assertThat(catalog.defaultWarehouseLocation(TABLE))
+        .startsWith(
+            LocationUtil.stripTrailingSlash(temp.toUri().toString())
```
Comment (ajantha-bhat) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:170]: This testcase was failing without the fix as `temp.toUri()` was having trailing slash which was leading to double slash in the path.

**Review by Fokko — APPROVED** [review]
LGTM

**Review by nastra — APPROVED** [review]

**Pull request merged**